File: ImplicitDetails/app/src/main/java/com/example/implicitdetails/MainActivity.java
package com.example.implicitdetails;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText url;
    private Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        url = findViewById(R.id.editText);
        btn = findViewById(R.id.button);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String urlText = url.getText().toString();
                Toast.makeText(MainActivity.this,urlText, Toast.LENGTH_SHORT).show();
                Uri webpage = Uri.parse(urlText);
                Intent intent = new Intent(Intent.ACTION_VIEW, webpage);
                startActivity(intent);
            }
        });
    }
}

File: Implicit_Intent/app/src/main/java/com/example/implicit_intent/MainActivity.kt
package com.example.implicit_intent

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mailBtn.setOnClickListener(View.OnClickListener {
            val mailId = editText.text
            val i = Intent()
            i.action = Intent.ACTION_SENDTO
            i.data = Uri.parse("mailto:${mailId}")
            startActivity(i)
        })

        browseBtn.setOnClickListener(View.OnClickListener {
            val addr = editText.text
            val i = Intent()
            i.action = Intent.ACTION_VIEW
            i.data = Uri.parse("http://$addr")
            startActivity(i)
        })

        dialBtn.setOnClickListener(View.OnClickListener {
            val num = editText.text.toString()
            val i = Intent()
            i.action = Intent.ACTION_DIAL
            i.data = Uri.parse("tel:$num")
            startActivity(i)
        })
    }
}